app/assets/javascripts/discourse/lib/render-tag.js.es6

```diff
@@ -3,7 +3,12 @@ export default function renderTag(tag, params) {
   tag = Handlebars.Utils.escapeExpression(tag);
   const classes = ['tag-' + tag, 'discourse-tag'];
   const tagName = params.tagName || "a";
-  const href = (tagName === "a" && !params.noHref) ? " href='" + Discourse.getURL("/tags/" + tag) + "' " : "";
+  let path;
+  if (tagName === "a" && !params.noHref) {
+    const current_user = Discourse.User.current();
+    path = params.isPrivateMessage && current_user && current_user.staff ? `/u/${current_user.username}/messages/tag/${tag}` : `/tags/${tag}`;
+  }
+  const href = path ? " href='" + Discourse.getURL(path) + "' " : "";
```

Comment:
We should use [template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) here as well.